.github/workflows/scripts/push-images.sh

```diff
@@ -1,20 +1,66 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# shellcheck shell=bash
+##
+## USAGE: __PROG__
+##
+## "__PROG__" publishes oci tarballs created with xbuild to a registry.
+##
+## Usage example(s):
+##   ./__PROG__ 1.5.2
+##   ./__PROG__ v1.5.2
+##   ./__PROG__ v1.5.2 -scratch
+##   ./__PROG__ refs/tags/v1.5.2
+##   ./__PROG__ refs/tags/v1.5.2 -scratch
+##
+## Commands
+## - ./__PROG__ <version> [image-variant]   pushes images to the registry using given version.
 
 set -e
 
-IMAGETAG="$1"
-if [ -z "$IMAGETAG" ]; then
-    echo "IMAGETAG not provided!" 1>&2
-    echo "Usage: push-images.sh IMAGETAG" 1>&2
-    exit 1
+function usage {
+  grep '^##' "$0" | sed -e 's/^##//' -e "s/__PROG__/$me/" >&2
+}
+
+me=$(basename "$0")
+
+version="$1"
+if [ -z "$version" ]; then
+  usage
+  echo -e "\n Errors:\n * the version must be provided."
+  exit 1
 fi
 
-echo "Pushing images tagged as $IMAGETAG..."
+# remove the git tag prefix
+# Push the images using the version tag (without the "v" prefix).
+# Also strips the refs/tags part if the GITHUB_REF variable is used.
+version="${version#refs/tags/v}"
+version="${version#v}"
+
+variant="$2"
+if [ -n "$variant" ] && [ "$variant" != "-scratch" ]; then
+  usage
+  echo -e "\n Errors:\n * The only supported variant is '-scratch'."
+  exit 1
+fi
+
+OCI_IMAGES=(
+  spire-server spire-agent k8s-workload-registrar oidc-discovery-provider
+)
+
+registry=gcr.io/spiffe-io
+if [ "$variant" = "-scratch" ] ; then
```

Comment:
Changes the registry to ghcr.io for scratch images to remove the need for duplicate script.